connector/authproxy/authproxy.go

```diff
@@ -69,7 +70,7 @@ func (m *callback) HandleCallback(s connector.Scopes, r *http.Request) (connecto
 	groups := m.groups
 	headerGroup := r.Header.Get(m.groupHeader)
 	if headerGroup != "" {
-		groups = append(groups, headerGroup)
+		groups = append(strings.Split(headerGroup, ","), groups...)
```

Comment:
```suggestion
		groups = append(strings.Split(headerGroup, ","), groups...)
```
I assume that trimming is also required, because `X-Groups: group-1, group2` is also a valid value.

Reply:
Cool 🙌, I cleaned the headerGroup to support spaces

Reply:
I added some tests using this [PR](https://github.com/dexidp/dex/pull/2379/files) as a guide